components/studentPortal/layout/Sidebar.tsx

```diff
@@ -22,45 +31,52 @@ interface SidebarItemProps {
   page: SidebarPage;
   icon: ReactElement;
   notifications?: boolean;
+  userRole: "freemium" | "student" | "coach";
 }
 const SidebarItem = ({
   name,
   link,
   page,
   icon,
   notifications,
+  userRole,
 }: SidebarItemProps) => {
   const { activePage } = useSelector(activePageSelector);
+  const isDisabled = userRole === "freemium";
```

Comment:
we should just pass isDisabled in as a prop. There's no need for the sidebarItem to know about userRoles. The component just needs to know if it should render itself as disabled or not

Reply:
done. used a state hook to set it in the Sidebar.tsx